src/main/java/org/folio/dcb/listener/kafka/CirculationEventListener.java

```diff
@@ -0,0 +1,81 @@
+package org.folio.dcb.listener.kafka;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.log4j.Log4j2;
+import org.folio.dcb.repository.TransactionRepository;
+import org.folio.dcb.service.LibraryService;
+import org.folio.spring.integration.XOkapiHeaders;
+import org.folio.spring.service.SystemUserScopedExecutionService;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.kafka.annotation.KafkaListener;
+import org.springframework.messaging.MessageHeaders;
+import org.springframework.stereotype.Component;
+
+import java.util.Objects;
+import java.util.UUID;
+
+import static org.folio.dcb.utils.TransactionHelper.getHeaderValue;
+import static org.folio.dcb.utils.TransactionHelper.parseEvent;
+
+@Log4j2
+@Component
+@RequiredArgsConstructor
+public class CirculationEventListener {
+  public static final String CHECK_IN_LISTENER_ID = "check-in-listener-id";
+  public static final String CHECK_OUT_LISTENER_ID = "loan-listener-id";
```

Comment:
I think this can be renamed to check_out_loan_listener_id just to avoid confusion that we are listening to a loan event closed after checkout.

Reply:
fixed